src/ir/stack-utils.h

```diff
@@ -158,6 +106,66 @@ struct StackSignature {
     return params == other.params && results == other.results &&
            unreachable == other.unreachable;
   }
+
+  // Whether a block whose contents have stack signature `a` could be typed with
+  // stack signature `b`, i.e. whether it could be used in a context that
+  // expects signature `b`. Formally:
+  //
+  // [t1*] -> [t2*] <: [s1* t1'*] -> [s2* t2'*] iff
+  //
+  //  - t1'_i <: t1_i
+  //  - t2_i <: t2'_i
+  //  - s1_i <: s2_i
```

Comment:
I think the subscripts make it clearer that we need to check element-wise subtyping and that the number of elements needs to match. It would be even clearer to write that out explicitly, though, so I'll do that and remove the messy-looking subscripts.